File: Crio/LinkedList/11_RotateList/RotateList.java
/*
public class ListNode {
   public int val; 
   public ListNode next;
   public ListNode(int x) { val = x; next = null; }
}
*/
import java.util.*;
import crio.ds.List.ListNode;

// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

class RotateList{
	public static void main(String[] args) {
		Scanner sc=new Scanner(System.in);
		int n=sc.nextInt();
		ArrayList<Integer> inp=new ArrayList<Integer>();
		for(int i=0;i<n;i++)
			inp.add(sc.nextInt());
		int k=sc.nextInt();
		sc.close();
		ListNode head=ListNode.createList(inp);
		Solution so=new Solution();
		head=so.rotateList(head,k);
		List<Integer> ans=ListNode.extractList(head);
		for(int i=0;i<ans.size();i++)
		{
			System.out.print(ans.get(i)+" ");
		}
	}
}

File: Crio/LinkedList/11_RotateList/RotateList.cpp
#include<bits/stdc++.h>
using namespace std;
#include "Solution.cpp"
#include "../crio/cpp/io/FastIO.hpp"

// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

int main() {


	FastIO();
	int n, k;
	cin >> n;
	vector<int> list(n);
	for ( int i = 0; i < n; i++)
		cin >> list[i];

	cin >> k;
	ListNode * head = ListNode().createList(list);

	head = rotateList(head, k);
	vector<int> resultList = ListNode().extractList(head);
	for ( int it : resultList) {
		cout << it << " ";
	}
}
